src/gluonts/dataset/artificial/_base.py

```diff
@@ -143,10 +157,16 @@ def __init__(
         self.is_promotions = is_promotions
         self.holidays = holidays
 
+        if isinstance(self.freq, Week):
+            self.freq = Week(
+                self.freq.n, weekday=pd.Timestamp(start).weekday()
+            )
+        self.start = cast(pd.Period, pd.Period(start, self.freq))
```

Comment:
Because pyright complained, because of this:

```py
class Period(PeriodMixin):
    ordinal: int  # int64_t
    freq: BaseOffset

    # error: "__new__" must return a class instance (got "Union[Period, NaTType]")
    def __new__(  # type: ignore[misc]
        cls,
        value=...,
        freq: int | str | BaseOffset | None = ...,
        ordinal: int | None = ...,
        year: int | None = ...,
        month: int | None = ...,
        quarter: int | None = ...,
        day: int | None = ...,
        hour: int | None = ...,
        minute: int | None = ...,
        second: int | None = ...,
    ) -> Period | NaTType: ...
```


https://github.com/pandas-dev/pandas/blob/9289c46e166533dfafd18843137f101624cf41a5/pandas/_libs/tslibs/period.pyi#L62-L79